src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java

```diff
@@ -426,6 +413,15 @@ public Integer getSshConnectionTimeOut() {
         return sshConnectionTimeOut;
     }
 
+    public void updateHostData() {
+        try {
+            hostsMap = getForemanAPI().getCompatibleHosts();
+        } catch (Exception e) {
+            LOGGER.error(e);
```

Comment:
We shouldn't use a stale data, what about to reset `hostsMap` to `null` in this case?